Update RC1 migration guide with migration to 1.0.0 content.

cesarbs opened the pull request:
cc @danroth27 @Rick-Anderson @rachelappel 

Fixes #1492

danroth27 commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,47 +28,50 @@ You must update the SDK version in ``global.json``, as this file is used to conf
-The EntityFramework.\* packages and namespaces are changing to Microsoft.EntityFrameworkCore.\*.
-All ASP.NET Core package versions are now 1.0.0-\*.
-Microsoft.Data.Entity.* is now Microsoft.EntityFrameworkCore.*
+All ``Microsoft.AspNet.*`` packages and namespaces have been renamed to ``Microsoft.AspNetCore.*``. The ``EntityFramework.*`` packages and namespaces have been renamed to ``Microsoft.EntityFrameworkCore.*``. ``Microsoft.Data.Entity.*`` is now ``Microsoft.EntityFrameworkCore.*``.
```
I actually like the bullets better in this case.

cesarbs replied:
You mean I should create bullet items for each of those?

danroth27 commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -1,22 +1,22 @@
-Migrating from ASP.NET 5 RC1 to ASP.NET Core
-============================================
+Migrating from ASP.NET 5 RC1 to ASP.NET Core 1.0
```
Migrating from ASP.NET Core RC1 or RC2 to ASP.NET Core 1.0

cesarbs replied:
That's a bit confusing. Migrating from RC2 involves almost no changes. Maybe I should write a topic on migrating from RC2?

cesarbs commented:
@Rick-Anderson This is the latest that I have.

Inviting reviewers: @JunTaoLuo @Tratcher @davidfowl @victorhurdugaci

JunTaoLuo commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+    }
+  }
+
+Update Target Framework Monikers (TFMs)
```
As I noted in the issue, this is covered by the document here: https://dotnet.github.io/docs/core-concepts/dnx-migration.html. Also, this is not exactly complete since library projects should target `netstandard1.x` and add dependencies to `NETStandard.Library` instead of `Microsoft.NETCore.App` in most cases.

victorhurdugaci commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -1,22 +1,22 @@
+Migrating from ASP.NET 5 RC1 to ASP.NET Core 1.0
+================================================
 
 By `Rachel Appel`_
```
Add your name :)

victorhurdugaci commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+Microsoft.AspNet.Tooling.Razor                Microsoft.AspNetCore.Razor.Tools
+==========================================    ===================================================
+
+All ASP.NET Core package versions are now ``1.0.0``. Tooling packages used in ASP.NET Core 1.0 applications should use version ``1.0.0-preview2-final``.
```
"Tooling" I think should be "Tools" as in "Tools packages"

victorhurdugaci commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+
+The ``commands`` section of ``project.json`` is no longer supported. Use ``dotnet run`` or ``dotnet <DLL name>`` instead.
+
+.NET Core CLI has introduced the concept of tools. ``project.json`` now supports a ``tools`` section where packages containing tools can be specified. Some important functionality for ASP.NET Core 1.0 applications has been moved to tools.
```
Is the branding ".net core cli" or just ".net cli"?

cesarbs replied:
Their docs say .NET Core CLI.

cesarbs replied:
I just removed it altogether.

victorhurdugaci commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+File watcher
+^^^^^^^^^^^^
+
+The ``watch`` command, formerly provided by the ``Microsoft.Dnx.Watcher`` package, has been replaced by the ``watch`` .NET Core CLI tool, available through the ``Microsoft.DotNet.Watcher.Tools`` package:
```
Add a reference to `dotnet-watch` https://github.com/aspnet/Docs/pull/1515

victorhurdugaci commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-All classes prefixed with WebApplication have been renamed to WebHost. This includes:
+Since ASP.NET Core 1.0 apps are just console apps, you must define an entry point for your application that sets up a web host and runs it. Below is an example from the startup code for the `Web Application` template in Visual Studio:
```
Isn't `Web Application <something>` ?

JunTaoLuo replied:
You create a AspNetCore Web Application project and then choose between EmptyWeb, WebAPI and WebApplication right?

victorhurdugaci commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -97,262 +246,282 @@ WebApplicationService          WebHostService
+=============================================  =============================================
+ASP.NET 5 RC1                                        ASP.NET Core 1.0
+=============================================  =============================================
+Microsoft.AspNet.Mvc.HttpOkObjectResult        Microsoft.AspNetCore.Mvc.OkObjectResult
```
Did we rename to ASP.NET Core before or after RC1?

cesarbs replied:
After.

JunTaoLuo commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+Publishing to IIS
+^^^^^^^^^^^^^^^^^
+
+IIS publishing is now provided via the ``publish-iis`` tool in the ``Microsoft.AspNetCore.Server.IISIntegration`` package. If you intend to run your app behind IIS, add the ``publish-iis`` tool to your ``project.json`` like this:
```
`publish-iis` is in the `Microsoft.AspNetCore.Server.IISIntegration.Tools` package not `Microsoft.AspNetCore.Server.IISIntegration`.

JunTaoLuo commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+    "tools": {
+      "Microsoft.AspNetCore.Server.IISIntegration.Tools": {
+        "version": "1.0.0-preview2-final",
+        "imports": "portable-net45+win8+dnxcore50"
```
I don't think the imports are needed anymore.

JunTaoLuo commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+
+All ASP.NET Core package versions are now ``1.0.0``. Tooling packages used in ASP.NET Core 1.0 applications should use version ``1.0.0-preview2-final``.
+
+Commands and tools
```
@danroth27, we spent some time previously to order the sections in order of relevance. I think we need to do that again with the newer changes. For example, in terms of migrating from RC1 -> RTM, I'm not sure if information regarding some of tools packages the should be above the webhost changes.

cesarbs replied:
Why not alphabetical :sob:

JunTaoLuo commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -28,64 +28,213 @@ You must update the SDK version in ``global.json``, as this file is used to conf
+      }
+  }
+
+Because ASP.NET Core 1.0 apps are now console apps, you must add the ``emitEntryPoint`` to the ``buildOptions`` section of your application's ``project.json``:
```
again this is covered in the dotnet migration doc, but I suppose it's important enough to duplicate the call out here. There are also other build options that may be relevant like `preserveCompilationContext` for MVC apps.

JunTaoLuo replied:
nvm, this is covered later in the MVC section.

JunTaoLuo commented on aspnet/migration/rc1-to-rc2.rst:
```diff
@@ -97,262 +246,282 @@ WebApplicationService          WebHostService
-	  stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" 
-	  forwardWindowsAuthToken="false"/>
+      <aspNetCore processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%"
+                  stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout"
```
Is it just me or does the tabs here seem a lot more than expected?

cesarbs replied:
I wanted to keep the args aligned.

cesarbs commented:
Just pushed my final updates. @Rick-Anderson will take it from here.

A few things to note:

* I've added a link to the `dotnet-watch` tutorial which will only work after https://github.com/aspnet/Docs/pull/1515 is merged. Until then Travis is broken.
* I've added a short document on RC2 to 1.0 migration. It's very small and maybe @danroth27 was right that we don't need it. I've left it like this for now though, and added a note to the document on migration from RC1 that people should check the other document for additional migration information.
* I've removed information that was already available in the .NET Core docs and linked were appropriate. *Some* of it thought I've left there since it's information somewhat specific to ASP.NET Core apps.
* I've added myself as one of the authors :smile: 

Paging some people from MVC who might catch stuff that I've missed: @NTaylorMullen @rynowak @ajaybhargavb @ryanbrandenburg @Eilon

Rick-Anderson commented:
@Eilon  @danroth27  The following link 
For a complete list of breaking changes in the ASP.NET Core RC2 , see `ASP.NET Core RC2 announcements <https://github.com/aspnet/announcements/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0.0-rc2>`_ 

filter in URL :  ``is:open is:issue milestone:1.0.0-rc2`` results in 3 pages. It includes announcements which are not breaking changes. Can I filter on **label:"Breaking change"**
``is:open is:issue milestone:1.0.0-rc2 label:"Breaking change" `` which gets it down to two pages?

I'll be wordsmithing this

Rick-Anderson commented:
@rowanmiller  will https://docs.efproject.net/en/latest/miscellaneous/rc1-rc2-upgrade.html be updated for RTM? Will the URL change?

Rick-Anderson commented on aspnet/migration/rc1-to-rtm.rst:
```diff
@@ -0,0 +1,483 @@
+
+The new :dn:class:`~Microsoft.AspNetCore.Mvc.ControllerAttribute` be used to mark a class and their subclasses as controllers. A classe whose name doesn't end in ``Controller`` and derives from a base class that ends in ``Controller`` is no longer considered a controller. In this scenario, :dn:class:`~Microsoft.AspNetCore.Mvc.ControllerAttribute` must be applied to the derived class itself or to  ...
+
+A type is not considered a controller if all the following conditions apply:
```
>A type is not considered a controller if all the following conditions apply: 

@rynowak  I'm removing not, I think this is backwards. 
A type is ~~not~~ considered a controller if all the following conditions apply:

Eilon commented:
@Rick-Anderson I would use the `is:open is:issue milestone:1.0.0-rc2` filter and just not say specifically "breaking changes." It's also of not a *complete* list because we both deliberately omit some things from there, and also of course inadvertently omit things as well. Rather, it's a "list of significant changes made during the XYZ milestone."

Eilon commented on aspnet/migration/rc2-to-rtm.rst:
```diff
@@ -0,0 +1,56 @@
+
+This migration guide covers migrating an ASP.NET Core RC2 application to ASP.NET Core 1.0.
+
+There weren't many significant changes to ASP.NET Core between the RC2 and 1.0 releases. For a complete list of changes, see the `ASP.NET Core 1.0 announcements <https://github.com/aspnet/announcements/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0.0>`_.
```
`For a list of *some of the most significant* changes` (it's definitely not a complete list)

Rick-Anderson replied:
fixed with
See the following resources for a list of some of the most significant changes, announcements and migrations information:

Eilon commented on aspnet/migration/rc2-to-rtm.rst:
```diff
@@ -0,0 +1,56 @@
+Tools
+-----
+
+You no longer need to use ``imports`` for tools in ``project.json``. You can change, for example, the following entry:
```
This is true only for the tools that we ship, it's certainly not generally true.

Eilon commented on aspnet/migration/rc2-to-rtm.rst:
```diff
@@ -0,0 +1,56 @@
+MVC
+---
+
+The :dn:class:`~Microsoft.AspNetCore.Html.HtmlString` class has been removed from MVC. You should now use the :dn:class:`~Microsoft.AspNetCore.Html.HtmlString` class provided in the ``Microsoft.AspNetCore.Html.Abstractions`` package, which was formerly named ``HtmlEncodedString``. If you were using :dn:class:`~Microsoft.AspNetCore.Html.HtmlString` directly in your Razor views, you must add a ``@u ...
```
I'm not sure it makes sense to use a "type" link to a type that no longer exists, because presumably it'll be a 404...

Rick-Anderson commented on aspnet/migration/rc1-to-rtm.rst:
```diff
@@ -0,0 +1,498 @@
+            "version": "1.0.0",
+            "type": "platform"
+          }
+        }
```
@Eilon @victorhurdugaci @JunTaoLuo  That's not what my 14.1.20624.0 build (latest) generates. This is what I get
```
"frameworks": {
    "netcoreapp1.0": {
      "imports": [
        "dotnet5.6",
        "portable-net45+win8"
      ]
    }
  },
```

cesarbs replied:
Yeah, it should contain the imports. My bad.

cesarbs replied:
But please leave the `"type": "platform"` there.

Rick-Anderson commented on aspnet/migration/rc1-to-rtm.rst:
```diff
@@ -0,0 +1,500 @@
+
+.. original -Delete ``<app name> - Web Deploy-publish.ps1``. This is a script generated by Visual Studio for web deploy. There is a version for ASP.NET 5 RC1 projects (which are DNX based) and a different script for ASP.NET Core 1.0 projects (which are dotnet based), and those are incompatible with each other. As such, when migrating to ASP.NET Core 1.0, you need to delete the old script and let  ...
+
+Delete any *<app name> - Web Deploy-publish.ps1* scripts created with Visual Studio web deploy using ASP.NET 5 RC1. The ASP.NET 5 RC1 scripts (which are DNX based) are not compatible with dotnet based scripts. Use Visual Studio to generate new web deploy scripts. 
```
@Eilon @victorhurdugaci @JunTaoLuo  total rewrite with original above commented out with 
``.. original``

cesarbs replied:
Nice, reads much better!

Rick-Anderson commented on aspnet/migration/rc1-to-rtm.rst:
```diff
@@ -0,0 +1,500 @@
+
+.. original -If ``applicationhost.config`` was created with ASP.NET 5 RC1 or an earlier release, in ASP.NET Core it will point to the wrong application folder. The ``applicationhost.config`` file will read ``wwwroot`` as the application folder and this is where IIS will look for the ``web.config`` file. But since the ``web.config`` file now goes in the ``approot``, IIS won't find the file and the ...
+
+An *applicationhost.config* file created with ASP.NET 5 RC1 will point ASP.NET Core to an invalid :ref:`content root <content-root-lbl>` location. With such a *applicationhost.config* file, ASP.NET Core will be configured with :ref:`content root <content-root-lbl>`/:ref:`web root <web-root-lbl>` as the :ref:`content root <content-root-lbl>` folder and therefore look for *web.config* in ``Content  ...
```
@Eilon @victorhurdugaci @JunTaoLuo total rewrite with original above commented out with  .. original

JunTaoLuo replied:
cc @moozzyk who is more familiar with this change.

rowanmiller commented:
@Rick-Anderson the link you want is https://docs.efproject.net/en/latest/miscellaneous/rc2-rtm-upgrade.html. It will be live tomorrow morning.

ryanbrandenburg commented on aspnet/migration/rc1-to-rtm.rst:
```diff
@@ -0,0 +1,503 @@
+ASP.NET 5 MVC compile views
+---------------------------
+
+To compile views, set the ``preserveCompilationContext`` option in *project.json* to preserve the compilation context, as shown here:
```
I don't know that "to preserve the compilation context" will mean much to most people. If we care that they know why `preserveCompilationContext` is the way to compile views then we might want a more detailed description of what it does.

Tratcher commented on aspnet/migration/rc1-to-rtm.rst:
```diff
@@ -0,0 +1,503 @@
+  :depth: 1
+
+
+ASP.NET 5 RC1 apps were based on the .NET Execution Environment (DNX) and made use of DNX specific features. ASP.NET Core 1.0 is based on .NET Core, so you must first migrate your application to the new .NET Core project model. See `migrating from DNX to .NET Core CLI <http://dotnet.github.io/docs/core-concepts/dnx-migration.html>`_ for more information.
```
bad link? https://docs.microsoft.com/en-us/dotnet/articles/core/migrating-from-dnx

JunTaoLuo replied:
Looks like it. PR: https://github.com/aspnet/Docs/pull/1542
